python/ray/data/_internal/logical/operators/one_to_one_operator.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from ray.data._internal.logical.interfaces import LogicalOperator
+
+
+class AbstractOneToOne(LogicalOperator):
+    """Abstract class for one-to-one logical operators should
+    be converted to their corresponding physical operators.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        input_op: Optional[LogicalOperator],
+    ):
+        """
+        Args:
+            name: Name for this operator. This is the name that will appear when
+                inspecting the logical plan of a Dataset.
+            input_op: The operator preceding this operator in the plan DAG. The outputs
+                of `input_op` will be the inputs to this operator.
+        """
+        super().__init__(name, [input_op] if input_op else [])
+
+    @property
+    def input_dependency(self) -> LogicalOperator:
+        return self._input_dependencies[0]
+
+    @property
```

Comment:
nit, use `abc.abstractmethod` instead of raising exceptions. because when using "abc", an error will be raised when instantiating the object, instead of when calling the method.